### PR TITLE
MNT Switch to new codecov uploader

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripeltd/product-issues/issues/441

Plan is to merge this as is and then trigger a travis build with codecov on it to validate it works

### CODE REVIEW

File: config/base.yml [hunk 1]
```diff
@@ -111,6 +111,16 @@ before_script:
   # The following line is used for testing the cms module itself
   - if [[ $BEHAT_TEST && $cms_installed == 0 ]]; then (vendor/bin/serve --bootstrap-file tests/behat/serve-bootstrap.php &> artifacts/serve.log &); fi
 
+  # codecov
+  # https://about.codecov.io/blog/introducing-codecovs-new-uploader/
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --import; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux.SHA256SUM; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux.SHA256SUM.sig; fi
+  - if [[ $PHPUNIT_COVERAGE_TEST ]]; then gpg --verify codecov-linux.SHA256SUM.sig codecov-linux.SHA256SUM; fi
```
Comment (chillu) [inline config/base.yml:120]: `gpg` and `shasum` aren't mentioned as included on xenial, I guess we'll find out when travis runs.
https://docs.travis-ci.com/user/reference/xenial/

**Comment by emteknetnz** [comment]
Triggered a build on silverstripe-config which only takes a few minutes to run

https://travis-ci.com/github/silverstripe/silverstripe-config/builds/228837846

**Comment by chillu** [comment]
Hah me too, sorry was too slow :D https://travis-ci.com/github/silverstripe/silverstripe-framework/builds/228383310

**Comment by emteknetnz** [comment]
it sort of ran but it didn't :/ https://travis-ci.com/github/silverstripe/silverstripe-config/jobs/513421204

**before_script**
```
0.21s$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --import; fi
gpg: key 806BB28AED779869: public key "Codecov Uploader (Codecov Uploader Verification Key) <security@codecov.io>" imported
gpg: Total number processed: 1
gpg:               imported: 1
before_script.46
0.44s$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux; fi
before_script.47
$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux.SHA256SUM; fi
before_script.48
$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then curl -Os https://uploader.codecov.io/latest/codecov-linux.SHA256SUM.sig; fi
before_script.49
$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then gpg --verify codecov-linux.SHA256SUM.sig codecov-linux.SHA256SUM; fi
gpg: Signature made Fri 11 Jun 2021 05:50:14 PM UTC
gpg:                using RSA key 27034E7FDB850E0BBC2C62FF806BB28AED779869
gpg: Good signature from "Codecov Uploader (Codecov Uploader Verification Key) <security@codecov.io>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 2703 4E7F DB85 0E0B BC2C  62FF 806B B28A ED77 9869
0.26s$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then shasum -a 256 -c codecov-linux.SHA256SUM; fi
codecov-linux: OK
before_script.51
0.00s$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then chmod +x codecov-linux; fi
```
**after_script**
```
$ if [[ $PHPUNIT_COVERAGE_TEST ]]; then ./codecov-linux -f coverage.xml; fi
     _____          _
    / ____|        | |
   | |     ___   __| | ___  ___ _____   __
   | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
   | |___| (_) | (_| |  __/ (_| (_) \ V /
    \_____\___/ \__,_|\___|\___\___/ \_/

  Codecov report uploader 0.1.0
=> Project root located at: /home/travis/build/silverstripe/silverstripe-config
Unable to detect service, please specify manually.
```

Looks like error comes from here https://github.com/codecov/uploader/blob/master/src/index.js#L186

It's not autodetecting that it's running on Travis CI?

**Comment by emteknetnz** [comment]
Very odd PR to test adding an env variable to get codecov to detect that it's running on travis ci https://github.com/silverstripe/silverstripe-config/pull/50

**Pull request merged**